chains/evm/cli/bridge/register-generic-resource.go

```diff
@@ -9,27 +17,56 @@ var registerGenericResourceCmd = &cobra.Command{
 	Use:   "register-generic-resource",
 	Short: "Register a generic resource ID",
 	Long:  "Register a resource ID with a contract address for a generic handler",
-	Run:   registerGenericResource,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		txFabric := evmtransaction.NewTransaction
+		return RegisterGenericResourceCmd(cmd, args, txFabric)
+	},
+}
+
+func BindRegisterGenericResourceCmdFlags(cmd *cobra.Command) {
+	cmd.Flags().String("handler", "", "handler contract address")
+	cmd.Flags().String("resourceId", "", "resource ID to query")
+	cmd.Flags().String("bridge", "", "bridge contract address")
+	cmd.Flags().String("target", "", "contract address to be registered")
+	cmd.Flags().String("deposit", "0x00000000", "deposit function signature")
+	cmd.Flags().String("execute", "0x00000000", "execute proposal function signature")
+	cmd.Flags().Bool("hash", false, "treat signature inputs as function prototype strings, hash and take the first 4 bytes")
+
+	err := cmd.MarkFlagRequired("handler")
```

Comment:
@tcar121293 Maybe this is something that can be wrapped in a separate function (in your rewrite), as we will have a lot of these snips trough out the codebase.